### DISCUSSION TITLE
SWC still compile even if interfaces are not fully implemented by classes

### DISCUSSION THREAD

**Discussion by ghost** [swc-project/swc#8051] [Q&A]
Well, the title speaks for itself: perhaps I can't find it in docs, but if I have something like this:
```typescript
interface IDo {
  do(): void;
}

class DoImpl implements IDo {}
```
No errors are shown on compilation, although I'll get `do is not a function` error if I'll try to call this function.
Why it is not complaining? Is there any way to enable this behaviour?

**Comment by kwonoj** [comment]
SWC doesn't have typechecking features, so this is natural since `implements` keyword works against non-runtime interface.

**Reply by ghost** [reply]
Oh, ok. Any tips on how to check types before compiling then?
